### PR TITLE
rec: Process policy and potential Drop action after Lua hooks

### PR DESCRIPTION
**Pull request by omoerbeek**
### Short description

If preresolve returns `true` and indicates the packet should be dropped, do it. 

### Checklist

I have:
- [X] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [X] compiled this code
- [X] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)
- [ ]  checked that this code was merged to master

### CODE REVIEW

File: pdns/pdns_recursor.cc [hunk 1]
```diff
@@ -1939,6 +1939,15 @@ static void startDoResolve(void *p)
         }
       }
     }
+    else if (t_pdl) {
```
Comment (omoerbeek) [inline pdns/pdns_recursor.cc:1942]: Maybe it would be better to rearrange a bit so this block ends up close to the `preresolve()` call?

**Review by rgacogne — APPROVED** [review]
Looks good. A regression test would be nice to have. I wonder if we should call `handlePolicyHit` after the `nodata` and `nxdomain` hooks as well?

File: pdns/pdns_recursor.cc (hunk 1 above)
Comment (omoerbeek) [reply pdns/pdns_recursor.cc:1942]: I tried, but it was too messy in the end.

**Comment by omoerbeek** [comment]
> Looks good. A regression test would be nice to have. I wonder if we should call `handlePolicyHit` after the `nodata` and `nxdomain` hooks as well?

I'll take a look at the other cases and ad regress test(s).

**Review by rgacogne — APPROVED** [review]

**Pull request merged**